Calculator_Project/Source.cpp

```diff
@@ -0,0 +1,251 @@
+#include <iostream>
+#include <vector>
+#include <iterator>
+#include <string>
+
+template <typename T>
+void vec_print(const std::vector<T>& container) {   //Print vector, entry by entry
+    for (const T equation : container)              //Found online, modified to generic type
+        std::cout << equation << ' ';
+    std::cout << std::endl;
+}
+
+bool char_is_num(const char c) {        //Return true if character c is a numeral, false if not
+    if ((c == '0') or (c == '1') or (c == '2') or (c == '3') or (c == '4') or (c == '5') or (c == '6') or (c == '7') or (c == '8') or (c == '9')) {
+        return true;
+    }
+    else {
+        return false;
+    }
+}
+
+
+int main() {
+
+    std::string equ_str;                    //Initialize equation variables, and take input of equation
+    std::cout << "Enter equation: ";
+    std::cin >> equ_str;
+    std::vector<std::string> equ_vec;
+
+    std::string num_str;                    //Variable to hold numeric characters, to combine multiple numeric characters into one number
+    int equ_str_index = 0;                  //Variable to track which character in the equation we are one
+    for (char c : equ_str) {                //Iterate through the equation, one character at a time

```

Comment:
This might be pedantic, but I would prefer a standard `for `loop (`for (int i = 0; i < equ_str.size(); i++)` rather than a `foreach` like what you have here when you end up needing to use an index variable anyways. It gets rid of the need to remember to increment the index, and `c` is only used in the `switch` anyways as far as I can see